lib/linguist/heuristics.rb

```diff
@@ -274,7 +274,7 @@ def call(data)
     end
 
     disambiguate ".pl" do |data|
-      if /^(use v6|(my )?class|module)/.match(data)
+      if /^(use v6|(my)? class|module)/.match(data) #class might be a variable name in Perl 5
```

Comment:
2015-12-13 11:57 GMT+01:00 Paul Chaignon <notifications@github.com>:

> In lib/linguist/heuristics.rb
> <https://github.com/github/linguist/pull/2709#discussion_r47443025>:
>
> > @@ -274,7 +274,7 @@ def call(data)
> >      end
> >
> >      disambiguate ".pl" do |data|
> > -      if /^(use v6|(my )?class|module)/.match(data)
> > +      if /^(use v6|(my)? class|module)/.match(data) #class might be a variable name in Perl 5
>
> What do you think of changing the regex to ^\s*(use v6|(my )?class|module)
> ?
>

I can check it, if you want. There are also some other changes
(Makefile.PL) and stuff that are being blocked because of this single
thing. I think that since Perl6 is about to be released any time now, it's
important to get it right.